### PR TITLE
feat: Create base interfaces and typing for control mobile mode

### PR DESCRIPTION
**Pull request by salarenko**
#### Please provide a brief summary of this pull request.
This PR:
- creates set of utils used for creation of mobile modes for form controls.
  - `MobileModeBase<T>` - base class which should be extended by implementations of mobile versions of form controls (like `MultiInputMobileComponent`).
  - `MobileMode` - interface describing minimum requirements to support mobile mode in form control. It should be implemented by form controls (like `MultiInputMobileComponent`).
  - `MobileModeToken` - token used to globally configure mobile mode of specified form control.
  - `MobileModeControlName` - enum used to target which component provided configuration should match.
  - `MobileModeConfig` - mobile mode config has been exteded with `dialogConfig`.
- Updates Mobile Multi Input component and the documentation to use created base classes and utils.
- Updates Mobile Select component and the documentation to use created base classes and utils.
- Updates Mobile Menu component and the documentation to use created base classes and utils.
- Creates Control Mobile Mode testing utils

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/utils/base-class/mobile-mode.class.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._onDestroy$.next();
+        this._onDestroy$.complete();
+    }
+
+    /** @hidden */
+    private _getMobileModeConfig(): MobileModeConfig {
+        if (this._component.mobileConfig) {
+            const injectedConfig = this._mobileModes.find(mode => mode.controlName === this.controlName);
```
Comment (salarenko) [inline libs/core/src/lib/utils/base-class/mobile-mode.class.ts:59]: Here it needs some work, but this is just an example of retrieving configuration token.

Comment (JKMarkowski) [reply libs/core/src/lib/utils/base-class/mobile-mode.class.ts:59]: Best thing will be adding some `if ()` with higher priority of `mobileConfig` from component over `config` from provider.

**Comment by InnaAtanasova** [comment]
@salarenko if it's a `feat` can you document the Breaking Changes?

**Comment by salarenko** [comment]
> @salarenko if it's a `feat` can you document the Breaking Changes?

I did

**Pull request merged**